src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx

```diff
@@ -3280,45 +3279,60 @@ it("send telemetry when users click on exposure chart free scan", async () => {
   );
 });
 
-it("does not display the CSAT survey banner on the dashboard", () => {
+it("does not display the CSAT survey banner on the dashboard tab “action needed” to Plus users", () => {
   const ComposedDashboard = composeStory(
     DashboardUsPremiumResolvedScanNoBreaches,
     Meta,
   );
-  render(<ComposedDashboard />);
+  render(<ComposedDashboard elapsedTimeInDaysSinceInitialScan={1} />);
 
   const answerButton = screen.queryByRole("button", {
     name: "Neutral",
   });
   expect(answerButton).not.toBeInTheDocument();
 });
 
-it("displays the CSAT survey banner on the dashboard", () => {
+it("does not display the CSAT survey banner to users who do not have automatic data removal enabled", async () => {
+  const user = userEvent.setup();
   const ComposedDashboard = composeStory(
-    DashboardUsPremiumResolvedScanNoBreaches,
+    DashboardUsPremiumEmptyScanNoBreaches,
     Meta,
   );
-  render(
-    <ComposedDashboard
-      elapsedTimeSinceInitialScan={CONST_DAY_MILLISECONDS * 91}
-    />,
+  render(<ComposedDashboard elapsedTimeInDaysSinceInitialScan={1} />);
+
+  const fixedTab = screen.getByText("Fixed");
+  await user.click(fixedTab);
+
+  const answerButton = screen.queryByRole("button", {
+    name: "Neutral",
+  });
+  expect(answerButton).not.toBeInTheDocument();
+});
+
+it("displays the CSAT survey banner to Plus users, after more than 90 days since their initial scan", async () => {
+  const user = userEvent.setup();
+  const ComposedDashboard = composeStory(
+    DashboardUsPremiumResolvedScanNoBreaches,
+    Meta,
   );
+  render(<ComposedDashboard elapsedTimeInDaysSinceInitialScan={91} />);
+
+  const fixedTab = screen.getByText("Fixed");
+  await user.click(fixedTab);
 
   const answerButton = screen.queryByRole("button", {
     name: "Neutral",
   });
   expect(answerButton).toBeInTheDocument();
 });
 
-it("displays the initial CSAT survey banner only on the ”fixed” tab of the dashboard to users with automatically fixed data brokers", async () => {
+it("displays the initial CSAT survey banner only to Plus users with automatically fixed data brokers", async () => {
```

Comment:
Could it be that we also need a test that verifies that it's _not_ shown for non-US and non-Plus users? Because I was just able to trigger it in the Storybook for those.

Reply:
Only Plus subscribers have an [initial (monitoring) scan](https://github.com/mozilla/blurts-server/pull/4421/files#diff-bf1c62d2593ab4fd9dd9216909e61714ea7c5af537fa25ef9f26f07e21a3f833R19-R22) (“initial” refers to the `onerep_scan_reason`). Free users only have a “manual” free scan.

Reply:
Ah right, so that value should never be set to anything for free users. Thanks for the clarification!